Add new spatial smoothing example to spectral_cube docs

kakirastern opened the pull request:
Fixes #547. 

To add a new spatial smoothing example with the approach using the `spatial_smooth` method. Also, attempt 

1.  To reverse the ordering of the terms "spatial" and "spectral" in L5 to reflect page organization. 
2. To fix some math rendering error coming from L66 of the same section in the "smooth.rst" file.

keflavich commented:
You need to rebase against master; there shouldn't be changes to astropy-helpers

keflavich commented on docs/smoothing.rst:
```diff
@@ -72,6 +72,28 @@ the error persists, to then slowly increase epsilon until a valid common beam is
+    from astropy.convolution import Gaussian2DKernel
+
+    cube0 = fits.open('/some_path/some_file.fits')
+    cube = SpectralCube.read(cube0)
```
The above two lines should be squashed; `SpectralCube.read` can read a FITS file directly.

keflavich commented on docs/smoothing.rst:
```diff
@@ -72,6 +72,28 @@ the error persists, to then slowly increase epsilon until a valid common beam is
+
+    cube0 = fits.open('/some_path/some_file.fits')
+    cube = SpectralCube.read(cube0)
+    kernel = Gaussian2DKernel(1)
```
use keyword arguments here, e.g. `x_stddev=1`.  Then you don't need to say nearly as much below

keflavich commented on docs/smoothing.rst:
```diff
@@ -72,6 +72,28 @@ the error persists, to then slowly increase epsilon until a valid common beam is
+``Gaussian2DKernel`` to change the standard deviation of the Gaussian in x 
+from the :math:`1` in the sample to something else within the reasonable limits say :math:`2.5`.
+For more information regarding that class please refer to the 
+`documentation at astropy <http://docs.astropy.org/en/stable/api/astropy.convolution.Gaussian2DKernel.html>`_.
```
Simplify this whole paragraph to, e.g.:
```
"`x_stddev` specifies the width of the `Gaussian kernel <http://docs.astropy.org/en/stable/api/astropy.convolution.Gaussian2DKernel.html>`_.
Any `astropy convolution <kernel http://docs.astropy.org/en/stable/convolution/kernels.html>`_ is acceptable."

keflavich commented on docs/smoothing.rst:
```diff
@@ -72,6 +72,28 @@ the error persists, to then slowly increase epsilon until a valid common beam is
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+There is an alternative way to spatially smooth a data cube, which is using the
+:meth:`~spectral_cube.SpectralCube.spatial_smooth` method. Example code snippets
+are as follows::
```
(1) there is only one code snippet below
(2) Instead of saying "example code snippet follows", say something about what the snippet does, i.e., "This is an example of how you do X":

kakirastern commented:
Funny, I did rebase with the command `git rebase -i spectral-cube/master <my-branch-name>`... but for some reason it couldn't get rid of the changes in the astropy-helpers. Let me see if there is another way to fix this meanwhile. Or, have I been doing it the wrong way?

kakirastern commented:
I tried manually updating once to the wrong version (v 2.0.8) first, then seemed to have succeeded in updating to the "correct" version to v 2.0.9 in effect getting rid of the undesired astropy-helpers changes... It's weird, because I had not done anything extra to effect the unwanted changes in astropy-helpers in the first place. Maybe I have missed some bigger picture. Will try to be more careful next time. Thanks for the tip!

kakirastern commented:
I have rebased again and dropped that redundant commit for manual update of astropy-helpers the first time around to clean up the PR. If there is any further issue please let know.

keflavich commented:
@kakirastern We fixed some of the CI build issues today, could you rebase again?
